src/luascript.cpp

```diff
@@ -17214,9 +17214,16 @@ int LuaScriptInterface::luaMoveEventRegister(lua_State* L)
 	if (moveevent) {
 		if ((moveevent->getEventType() == MOVE_EVENT_EQUIP || moveevent->getEventType() == MOVE_EVENT_DEEQUIP) &&
 		    moveevent->getSlot() == SLOTP_WHEREEVER) {
-			uint32_t id = moveevent->getItemIdRange().at(0);
-			ItemType& it = Item::items.getItemType(id);
-			moveevent->setSlot(it.slotPosition);
+			std::vector<uint16_t> slots;
+			for (const auto id : moveevent->getItemIdRange()) {
+				ItemType& it = Item::items.getItemType(id);
+				uint16_t itemSlot = it.slotPosition;
+
+				if (std::find(slots.begin(), slots.end(), itemSlot) != slots.end()) {
```

Comment:
This is very inefficient. There's a very limited number of slots, you can track using a `bitset` instead